contracts/src/NStates.sol

```diff
@@ -80,46 +80,58 @@ contract NStates is IncrementalMerkleTree {
      */
     function move(
         uint256[7] memory pubSignals,
-        uint256[2] memory a,
-        uint256[2][2] memory b,
-        uint256[2] memory c
+        uint256[8] memory formattedProof,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
     ) public {
-        uint256 root = pubSignals[0];
-        uint256 troopInterval = pubSignals[1];
-        uint256 waterInterval = pubSignals[2];
-        uint256 hUFrom = pubSignals[3];
-        uint256 hUTo = pubSignals[4];
-        uint256 rhoFrom = pubSignals[5];
-        uint256 rhoTo = pubSignals[6];
-
-        require(rootHistory[root], "Root must be in root history");
+        // root = pubSignals[0];
```

Comment:
i like leaving these comments in better 

having the names rather than pubSignals[n] for the rest of the func is less error prone & better hygiene

Reply:
unfortunately I tried using local variables but I got that solidity compiler error for it. major pain. there's a flag --via-ir that helps with this but it changes something about the build. do you think we should start using it?

Reply:
i see what you mean. totally didn't know about this solidity limitation- thanks for pointing it out 

what are your thoughts on using a MoveInputs struct here? https://medium.com/1milliondevs/compilererror-stack-too-deep-try-removing-local-variables-solved-a6bcecc16231. so we don't need to mess with compilation flags